src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5571,7 +5678,7 @@ private BoundExpression BindUnexpectedComplexElementInitializer(InitializerExpre
         }
 
         private BoundExpression BindCollectionInitializerElementAddMethod(
-            ExpressionSyntax elementInitializer,
+            SyntaxNode elementInitializer,
```

Comment:
nit: considering asserting hte types this is ok to be.

Reply:
This change has been reverted.